include/numerics/type_vector.h

```diff
@@ -1131,6 +1131,18 @@ TypeVector<T> TypeVector<T>::unit() const
 #endif
 
 }
+
+template <typename T>
+struct CompareTypes<TypeTensor<T>, TypeTensor<T>>
+{
+  typedef TypeTensor<T> supertype;
+};
+
+template <typename T, typename T2>
+struct CompareTypes<TypeTensor<T>, TypeTensor<T2>>
+{
+  typedef TypeTensor<typename CompareTypes<T,T2>::supertype> supertype;
+};
```

Comment:
Omg, I guess I never actually do addition/subtraction of `TypeVectors`. What a terrible typo